pkg/api/universal/conversation.go

```diff
@@ -55,35 +104,29 @@ func (a *Universal) ConverseAlpha1(ctx context.Context, req *runtimev1pb.Convers
 		return nil, err
 	}
 
-	scrubber, err := piiscrubber.NewDefaultScrubber()
-	if err != nil {
-		err := messages.ErrConversationMissingInputs.WithFormat(req.GetName())
-		a.logger.Debug(err)
-		return &runtimev1pb.ConversationResponse{}, err
-	}
-
-	for _, i := range req.GetInputs() {
-		msg := i.GetContent()
-
-		if i.GetScrubPII() {
-			scrubbed, sErr := scrubber.ScrubTexts([]string{i.GetContent()})
-			if sErr != nil {
-				sErr = messages.ErrConversationInvoke.WithFormat(req.GetName(), sErr.Error())
-				a.logger.Debug(sErr)
-				return &runtimev1pb.ConversationResponse{}, sErr
-			}
+	// prepare scrubber in case PII scrubbing is enabled (either at request level or input level)
+	var scrubber piiscrubber.Scrubber
 
-			msg = scrubbed[0]
-		}
+	// Check if PII scrubbing is needed for inputs or outputs
+	needsInputPIIScrubbing := needsInputScrubber(req)
+	needsOutputPIIScrubbing := needsOutputScrubber(req)
 
-		c := conversation.ConversationInput{
-			Message: msg,
-			Role:    conversation.Role(i.GetRole()),
+	if needsInputPIIScrubbing || needsOutputPIIScrubbing {
+		scrubber, err = piiscrubber.NewDefaultScrubber()
+		if err != nil {
+			err := messages.ErrConversationMissingInputs.WithFormat(req.GetName())
+			a.logger.Debug(err)
+			return &runtimev1pb.ConversationResponse{}, err
 		}
```

Comment:
We have lost the error contents here. I assume the original error was meant to be passed to the `Debug` ?

Please use Error log level, not debug.